### PR TITLE
Add request validation utilities

### PR DESCRIPTION
**Pull request by davejlong**
Based on [Twilio's security docs](https://www.twilio.com/docs/api/security), this PR adds 2 utility modules for validating that incoming requests into an application are coming from Twilio.

The `RequestValidator` module is a self-contained utility to verify that the signature generated from a URL and parameter map match a specified signature based on how Twilio builds their signatures.

The `RequestValidatorPlug` module is a `Plug` that can be used in `Plug.Router` or `Phoenix` to automatically verify requests and halt them returning an HTTP 401 status if the request is not received from Twilio.

### CODE REVIEW

**Comment by davejlong** [comment]
Credo was failing on my laptop, but I think that the failure might be a bug in Credo.

File: lib/ex_twilio/utilities/request_validator_plug.ex [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  def init(opts), do: opts
+
+  def call(conn, _opts) do
+    case conn |> get_req_header("x-twilio-signature") |> List.first do
+      nil -> conn |> send_resp(401, "Not authorized") |> halt
+      signature -> conn
+                   |> url_from_conn
+                   |> RequestValidator.validate(conn.params, signature)
+                   |> if(do: conn, else: conn |> send_resp(401, "Not authroized") |> halt)
```
Comment (zolrath) [inline lib/ex_twilio/utilities/request_validator_plug.ex]: authorized is misspelled as authroized

**Comment by danielberkompas** [comment]
@davejlong I can fix the credo warning, but I can't push to your branch. Can you enable that? Should have been an option for that in GitHub when you created the PR.

**Comment by davejlong** [comment]
@danielberkompas Not sure why you can't edit. The PR does allow edits from maintainers. On another note, I need to dive deeper into the logic. I'm having trouble getting it to work with Twilio, even though it works with the Ruby library.

**Comment by davejlong** [comment]
I have a new PR to submit this under. Closing for now.

**Pull request closed without merging**